experimental/dds/tree2/src/feature-libraries/typed-schema/typedTreeSchema.ts

```diff
@@ -80,14 +83,46 @@ export class TreeSchema<
 			Assume<T, TreeSchemaSpecification>["structFields"]
 		>(this.info.structFields);
 		this.structFields = objectToMapTyped(this.structFieldsObject);
-		this.mapFields = this.info.mapFields;
+		this.mapFields = this.info.mapFields as WithDefault<
+			Assume<T, TreeSchemaSpecification>["mapFields"],
+			undefined
+		>;
 		this.leafValue = this.info.leafValue as WithDefault<
 			Assume<T, TreeSchemaSpecification>["leafValue"],
 			undefined
 		>;
 	}
 }
 
+export type MapSchema = TreeSchema & MapSchemaSpecification;
+export type LeafSchema = TreeSchema & LeafSchemaSpecification;
+
+// TODO: this includes FieldNodeSchema when it shouldn't
+export type StructSchema = TreeSchema & {
+	[P in keyof (MapSchemaSpecification & LeafSchemaSpecification)]?: undefined;
+};
+// TODO: Split TreeSchema into union of node schema interface types, and make FieldNodeSchema not subset of StructSchema
+// Then replace bellow type checks with instance of tests.
+export type FieldNodeSchema = StructSchema & {
+	structFieldsObject: { [""]: FieldSchema };
```

Comment:
It isn't immediately clear to me what the empty prop name here represents. Can we document this?

Reply:
Done.